pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -338,7 +338,38 @@ public CompletableFuture<Void> initialize() {
             if (cursor.getName().startsWith(replicatorPrefix)) {
                 String localCluster = brokerService.pulsar().getConfiguration().getClusterName();
                 String remoteCluster = PersistentReplicator.getRemoteCluster(cursor.getName());
-                futures.add(addReplicationCluster(remoteCluster, cursor, localCluster));
+                final CompletableFuture<Void> future = new CompletableFuture<>();
+                addReplicationCluster(remoteCluster, cursor, localCluster).whenComplete((__, e) -> {
+                    if (e == null) {
+                        future.complete(null);
+                    } else {
+                        Throwable throwable = e;
+                        while (throwable.getCause() != null) {
+                            throwable = throwable.getCause();
+                        }
+                        if (throwable instanceof MetadataStoreException.NotFoundException
+                                && throwable.getMessage().equals(remoteCluster)) {
+                            log.warn("[{}] Remote cluster '{}' is not found while there is a replicator cursor,"
+                                            + " remove cursor '{}'", topic, remoteCluster, cursor.getName());
+                            ledger.asyncDeleteCursor(cursor.getName(), new DeleteCursorCallback() {
```

Comment:
How do we make sure we don't delete by mistake?
1. If there is a situation such as jitter, which makes the data temporarily unobtainable, will there be accidental deletion?
2. Will cursors be lost if the remote cluster is briefly unavailable?

Reply:
Only when `BrokerService#getReplication` accepts an empty `clusterDataOp` will we remove the cursor. The data is retrieved from:

```java
brokerService.pulsar().getPulsarResources().getClusterResources()
                        .getClusterAsync(remoteCluster)
```

ZK is always CP, so I think it won't be temporariy unobtainable like the network loss.

Reply:
> How do we make sure we don't delete by mistake?

+1

And this [suggestion](https://github.com/apache/pulsar/pull/19972#discussion_r1154048927) can avoid this mistake.

Reply:
Are these metadata non-persistent ZK nodes or persistent ZK nodes? 
If it is a non-persistent node, the short-term offline of the cluster will make the metadata disappear

Reply:
> Are these metadata non-persistent ZK nodes or persistent ZK nodes?

I think I need some time to evaluate it again. For example, I cannot answer this question from the MetadataStore API layer.

Reply:
I just have a quick offline discussion with @poorbarcode. I will avoid this check for metadata store in my latest commit.